### PR TITLE
Added blend function with opacity as parameter.

### PR DESCRIPTION
**Pull request by emilwestergren**

### CODE REVIEW

File: source/draw/gpu/GpuSurface.ooc [hunk 1]
```diff
@@ -81,10 +81,10 @@ GpuSurface: abstract class {
 	_bind: virtual func
 	_unbind: virtual func
 	clear: abstract func
+	blend: func (factor: Float) { this _context blend(factor) }
```
Comment (simonmika) [inline source/draw/gpu/GpuSurface.ooc]: Blend should be a state on the surface.

Comment (emilwestergren) [inline source/draw/gpu/GpuSurface.ooc]: OK, I just copied the code from the C# version of Kean where it is a function. Also by doing it this way I didn't have to adapt anything in Vidproc and the old blend hack for monochromes still works. So I thought it was a pretty clean solution for now but I can fix everything in a larger PR if you want me to.

**Comment by emilwestergren** [comment]
Updated.

**Comment by emilwestergren** [comment]
Updated.

File: source/draw/gpu/GpuContext.ooc [hunk 2]
```diff
@@ -56,6 +56,7 @@ GpuContext: abstract class {
 	drawPoints: abstract func (pointList: VectorList<FloatPoint2D>, transform: FloatTransform3D)
 	drawQuad: abstract func
 	enableBlend: abstract func (blend: Bool)
+	blend: abstract func (factor: Float)
```
Comment (simonmika) [inline source/draw/gpu/GpuContext.ooc]: Is GpuContext accessable outside this compontent?

Comment (emilwestergren) [inline source/draw/gpu/GpuContext.ooc]: Yes

Comment (simonmika) [inline source/draw/gpu/GpuContext.ooc]: Then this should not be here. It should be in the backend not in the frontend. The same goes for the other methods as well.

Comment (emilwestergren) [inline source/draw/gpu/GpuContext.ooc]: True

**Comment by emilwestergren** [comment]
Updated.

**Pull request merged**